skops/io/tests/test_external.py

```diff
@@ -0,0 +1,325 @@
+import sys
+
+import pytest
+from sklearn.datasets import make_classification, make_regression
+from sklearn.utils._testing import assert_allclose_dense_sparse
+
+from skops.io import dumps, loads
+from skops.io.tests._utils import assert_params_equal
+
+ATOL = 1e-6 if sys.platform == "darwin" else 1e-7
+# Default settings for generated data
+N_SAMPLES = 30
+N_FEATURES = 10
+N_CLASSES = 4  # for classification only
+
+
+@pytest.fixture(scope="module")
+def clf_data():
+    X, y = make_classification(
+        n_samples=N_SAMPLES,
+        n_classes=N_CLASSES,
+        n_features=N_FEATURES,
+        random_state=0,
+        n_redundant=1,
+        n_informative=N_FEATURES - 1,
+    )
+    return X, y
+
+
+@pytest.fixture(scope="module")
+def regr_data():
+    X, y = make_regression(n_samples=N_SAMPLES, n_features=N_FEATURES, random_state=0)
+    return X, y
+
+
+@pytest.fixture(scope="module")
+def rank_data(clf_data):
+    X, y = clf_data
+    group = [10 for _ in range(N_SAMPLES // 10)]
+    n = sum(group)
+    if N_SAMPLES > n:
+        group[-1] += N_SAMPLES - n
+    assert sum(group) == N_SAMPLES
+    return X, y, group
+
+
+def check_estimator(estimator, trusted):
+    loaded = loads(dumps(estimator), trusted=trusted)
+    assert_params_equal(estimator.get_params(), loaded.get_params())
+
+
+def check_method_outputs(estimator, X, trusted):
```

Comment:
isn't this common between these tests and scikit-learn ones? Would be nice to avoid duplication as much as we can here.